x/nodes/keeper/valStateChanges_test.go

```diff
@@ -350,6 +352,129 @@ func TestValidatorStateChange_EditAndValidateStakeValidatorAfterNonCustodialUpgr
+	assert.Nil(t, runStake(ctx, k, msgTemplate, operatorAcc2))
+
+	// Verify staked nodes having nil output address
+	validatorCur, found := k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.Nil(t, validatorCur.OutputAddress)
+	validatorCur, found = k.GetValidator(ctx, operatorAddr2)
+	assert.True(t, found)
+	assert.Nil(t, validatorCur.OutputAddress)
+
+	// Enable NCUST
+	codec.UpgradeFeatureMap[codec.NonCustodialUpdateKey] = -1
+
+	// Set an output address on Addr1 after NCUST
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = outputAddress
+	assert.Nil(t, runStake(ctx, k, msgTemplate, operatorAcc1))
+	validatorCur, found = k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.True(t, validatorCur.OutputAddress.Equals(outputAddress))
+
+	// Attempt to change the output address with operator's signature --> Fail
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = operatorAddr1
+	err := runStake(ctx, k, msgTemplate, operatorAcc1)
+	assert.Equal(t, k.codespace, err.Codespace())
+	assert.Equal(t, types.CodeUnequalOutputAddr, err.Code())
+
+	// Attempt to change the output address with output's signature --> Fail
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = operatorAddr1
+	err = runStake(ctx, k, msgTemplate, outputAccount)
+	assert.Equal(t, k.codespace, err.Codespace())
+	assert.Equal(t, types.CodeUnauthorizedSigner, err.Code())
+
+	// Enable OEDIT
+	codec.UpgradeFeatureMap[codec.OutputAddressEditKey] = -1
+
+	// Attempt to change the output address with operator's signature --> Fail
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = operatorAddr1
+	err = runStake(ctx, k, msgTemplate, operatorAcc1)
+	assert.Equal(t, k.codespace, err.Codespace())
+	assert.Equal(t, types.CodeDisallowedOutputAddressChange, err.Code())
+
+	// Attempt to change the output address with output's signature --> Success
+	msgTemplate.PublicKey = operatorAcc1
+	msgTemplate.Output = operatorAddr1
+	err = runStake(ctx, k, msgTemplate, outputAccount)
+	assert.Nil(t, err)
+	validatorCur, found = k.GetValidator(ctx, operatorAddr1)
+	assert.True(t, found)
+	assert.True(t, validatorCur.OutputAddress.Equals(operatorAddr1))
+
+	// Attempt to change the output address from nil
+	// with operator's signature --> Success
+	msgTemplate.PublicKey = operatorAcc2
+	msgTemplate.Output = outputAddress
+	err = runStake(ctx, k, msgTemplate, operatorAcc2)
+	assert.Nil(t, err)
+	validatorCur, found = k.GetValidator(ctx, operatorAddr2)
+	assert.True(t, found)
+	assert.True(t, validatorCur.OutputAddress.Equals(outputAddress))
+
+	// New stake acc3 with outputAccount with output's signature --> Success
+	msgTemplate.PublicKey = operatorAcc3
+	msgTemplate.Output = outputAddress
+	err = runStake(ctx, k, msgTemplate, outputAccount)
+	assert.Nil(t, err)
+	validatorCur, found = k.GetValidator(ctx, operatorAddr3)
+	assert.True(t, found)
+	assert.True(t, validatorCur.OutputAddress.Equals(outputAddress))
+}
```

Comment:
Can you please update the github description with:
1. A list of all the changes
2. Example `pocket` commands you ran to verify this on a locally running node
3. Just two sentences on a reasoning / rationale that we'll include in the release notes